### PR TITLE
feat(Link): support refs

### LINKED ISSUE #152: support refs in the `<Link />` component
@ryomtoob was working on using the `<Link />` component within a tablist which needed refs...we ended up converting the cauldron-react `<Link />` to a plain old `<a />` but this did point out that we need to support the usecase of having an element ref handle on the link

### PR DESCRIPTION
**Pull request by schne324**
There wasn't a demo page for `<Link />` so I [created one](https://deploy-preview-243--cauldron-react.netlify.com/components/link). I also added some unit tests 🔭 

Closes #152

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**
- [x] Follows the commit message policy, appropriate for next version
- [x] Tested for accessibility
- [x] Code is reviewed for security

### CODE REVIEW

File: src/components/Link/index.js [hunk 1]
```diff
@@ -1,14 +1,26 @@
...
+        {...other}
+      >
+        {children}
+      </a>
+    );
+  }
 }
 
-Link.propTypes = {
-  children: PropTypes.node.isRequired
```
Comment (schne324) [inline src/components/Link/index.js:13]: **NOTE**: `children` is no longer a required prop in order to support usage such as `<Link aria-label="Foo" className="icon-foo" />`

File: src/components/Link/index.js [hunk 2]
```diff
@@ -1,14 +1,26 @@
...
 import PropTypes from 'prop-types';
+import classNames from 'classnames';
 
-export default function Link({ children, ...other }) {
-  return (
-    <a className="dqpl-link" {...other}>
-      {children}
-    </a>
-  );
+export default class Link extends Component {
```
Comment (scurker) [inline src/components/Link/index.js]: Why the change from `function` to `class`?

Comment (schne324) [reply src/components/Link/index.js]: good catch, I didn't mean to leave this change in

**Review by scurker — APPROVED** [review]
